### PR TITLE
Complex monitoring: fixed bugs and added monitoring pages for student and lecturer and fix bug with duplicate

### PR DESCRIPTION
**Pull request by zhur4k**

### CODE REVIEW

File: modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      <button class="invisible" mat-icon-button disabled></button>
+      <mat-icon class="expandable-external folder_file_icons">
+        insert_drive_file
+      </mat-icon>
+      {{ node.Name }}   
+      <span class="time-container" *ngIf="node.Estimated > 0" [ngStyle]="{'color': converterService.getColorByTime(node.WatchingTime, node.Estimated)}">
+        &nbsp;({{ converterService.getStrTime(node.WatchingTime) }})
+        <span class="expected-time">
+          <span *ngIf="node.Estimated > 0" [ngStyle]="{'color': '#363636'}">
+            &nbsp;Ожидаемое время - {{ converterService.getStrTime(node.Estimated) }},
```
Comment (hzhurauski) [inline modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.html]: Переводы?

File: modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.less [hunk 2]
```diff
@@ -0,0 +1,153 @@
+.time-container {
+  position: relative;
+  display: inline-block;
+}
+
+.expected-time {
+  display: none;
+  position: fixed; /* Использование fixed позиционирования */
```
Comment (hzhurauski) [inline modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.less]: такие комменты лучше не писать) итак очевидно

File: modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            this.showLoader = false
+            this.treeControl.expandAll()
+        },
+        error => {
+            console.error('Ошибка при получении данных:', error);
+        }
+    );
+}
+
+hasChild = (_: number, node: any) => node.Children && node.Children.length > 0;
```
Comment (hzhurauski) [inline modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.ts]: давай без ```any```

File: modules/complex/src/app/service/complex.service.ts [hunk 4]
```diff
@@ -51,6 +52,16 @@ export class ComplexService {
       )
       .pipe(map((res) => res['Concept']))
   }
+  
+  public getStudentComplexMonitoringInfo(comlexId: string, studentId: string): Observable<ComplexStudentMonitoring> {
+    return this.http
+      .get<ComplexStudentMonitoring>(
+        '/Services/Concept/ConceptService.svc/GetStudentMonitoringInfo?complexId=' +
```
Comment (hzhurauski) [inline modules/complex/src/app/service/complex.service.ts]: такие пути лучше выносить в константы

File: modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.ts (hunk 3 above)
Comment (zhur4k) [reply modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.ts]: Исправил

File: modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.less (hunk 2 above)
Comment (zhur4k) [reply modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.less]: Исправил

File: modules/complex/src/app/service/complex.service.ts (hunk 4 above)
Comment (zhur4k) [reply modules/complex/src/app/service/complex.service.ts]: Сделано

**Review by hzhurauski — APPROVED** [review]

File: api/LMPlatform.Models/Concept.cs [hunk 5]
```diff
@@ -12,6 +12,51 @@ public class ConceptQuestions: ModelBase
         public virtual Concept Concept { get; set; }
         public virtual Question Question { get; set; }
     }
+    public class ConceptMonitoring : ModelBase
```
Comment (hawkrai) [inline api/LMPlatform.Models/Concept.cs]: Need create a separate file for each class

File: api/LMPlatform.Models/Concept.cs [hunk 6]
```diff
@@ -12,6 +12,51 @@ public class ConceptQuestions: ModelBase
...
+    {
+        public ConceptMonitoring()
```
Comment (hawkrai) [inline api/LMPlatform.Models/Concept.cs]: Do we have sql script for a new model?

**Review by hawkrai — CHANGES_REQUESTED** [review]

File: api/LMPlatform.Models/Concept.cs (hunk 6 above)
Comment (zhur4k) [reply api/LMPlatform.Models/Concept.cs]: It is dto

File: api/LMPlatform.Models/Concept.cs (hunk 5 above)
Comment (zhur4k) [reply api/LMPlatform.Models/Concept.cs]: https://github.com/hawkrai/CATSdesigner/pull/668/commits/612402fff3a8002ff5117220c28ca3080da97cd0

File: modules/complex/src/app/service/complex.service.ts (hunk 4 above)
Comment (zhur4k) [reply modules/complex/src/app/service/complex.service.ts]: https://github.com/hawkrai/CATSdesigner/pull/668/commits/bc00abd2689ae5de0dcadf3e1093cd2b68a3b9af

File: modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.less (hunk 2 above)
Comment (zhur4k) [reply modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.less]: https://github.com/hawkrai/CATSdesigner/pull/668/commits/bc00abd2689ae5de0dcadf3e1093cd2b68a3b9af

File: modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.html (hunk 1 above)
Comment (zhur4k) [reply modules/complex/src/app/complexGrid/components/monitoring-tree/monitoring-tree.component.html]: https://github.com/hawkrai/CATSdesigner/pull/668/commits/1b1a59f7baa96a7d8b8d91f1fb605781576e8e9c

**Pull request merged**